src/bio2bel/sources/biogrid.py

```diff
@@ -0,0 +1,257 @@
+    :param prefix: prefix to filter for (e.g. 'pubmed')
+    :param list_ids: list of identifiers
+    :return: filtered list of lists of ids
+    """
+    final_list = []
+    for ids in list_ids:
+        id_list = ids.split(separator)
+        flag = False
+        row_list = []
+        for i in id_list:
+            if i.startswith(prefix):
+                row_list.append(i)
+                flag = True
+        # if no matching value with prefix is found, append 'nan' as individual value
+        if not flag:
+            final_list.append('nan')
+        # if at least one value is found, append list
+        if len(row_list) > 0:
+            final_list.append(row_list)
+    return final_list
+
+
+def expand_df(df: pd.DataFrame, column_name: str) -> pd.DataFrame:
+    """Add row with same content to dataframe if there are multiple values in one column.
+
+    The specified column must contain an Iterable.
+
+    :param column_name: name of column with multiple values to be simplified
+    :param df: dataframe with multiple values in one column
+    :return: dataframe with expanded values
+    """
+    # identify columns with single values that do not neet to be expanded
+    columns_to_keep = df.columns.to_list()
+    columns_to_keep.remove(column_name)
+
+    # create expanded df
+    lens = [len(item) for item in df[column_name]]
+    df_to_keep = {column: np.repeat(df[column].values, lens) for column in columns_to_keep}
+
+    merged_df = {**df_to_keep, **{column_name: np.concatenate(df[column_name].values)}}
+    return pd.DataFrame(merged_df)
+
+
+def get_processed_biogrid() -> pd.DataFrame:
+    """Load BioGRDID file, filter and rename columns and return a dataframe.
+
+    :return: dataframe of preprocessed BioGRID data
+    """
+    df = _get_my_df()
+    # rename columns
+    df = df.rename(columns=BIOGRID_COLUMN_MAPPER)
+
+    # take relevant columns for source, target, alternative ids, relation and PubMed ID
+    df = df[[SOURCE, TARGET, ALT_SOURCE_ID, ALT_TARGET_ID, RELATION, PUBMED_ID]]
+
+    # filter for uniprot
+    df[SOURCE] = filter_for_prefix_single(list_ids=df[SOURCE], prefix=UNIPROT)
+    df[TARGET] = filter_for_prefix_single(list_ids=df[TARGET], prefix=UNIPROT)
+    # also in alternative ids
+    df[ALT_SOURCE_ID] = filter_for_prefix_multi(list_ids=df[ALT_SOURCE_ID], prefix=UNIPROT)
+    df[ALT_TARGET_ID] = filter_for_prefix_multi(list_ids=df[ALT_TARGET_ID], prefix=UNIPROT)
+
+    # drop rows if no uniprot id
+    df = df.dropna(subset=[SOURCE, ALT_SOURCE_ID], how='all')
+    df = df.dropna(subset=[TARGET, ALT_TARGET_ID], how='all')
+
+    # change uniprot/swiss-prot prefix to uniprot
+    df = df.replace(r'^uniprot/swissprot:.*', r'uniprot:.*', regex=True)
+
+    # expand dataframe if multiple uniprot ids exist
+    df = expand_df(df=df, column_name=ALT_SOURCE_ID)
+    df = expand_df(df=df, column_name=ALT_TARGET_ID)
+
+    # filter for relation
+    df[RELATION] = filter_for_prefix_single(
```

Comment:
Lots of unnecessary code here for normalizing the data.From

```bash
cat BIOGRID-ALL-3.5.183.mitab.txt | cut -f 12 | sort | uniq -c
   1 Interaction Types
73125 psi-mi:"MI:0403"(colocalization)
247365 psi-mi:"MI:0407"(direct interaction)
56701 psi-mi:"MI:0794"(synthetic genetic interaction defined by inequality)
217441 psi-mi:"MI:0796"(suppressive genetic interaction defined by inequality)
563593 psi-mi:"MI:0799"(additive genetic interaction defined by inequality)
10562 psi-mi:"MI:0914"(association)
586598 psi-mi:"MI:0915"(physical association)
```

You can have exact strings for these things in your code rather than mapping through an arbitrary intermediate

Reply:
So the `filter_for_prefix_single` is unnecessary

Reply:
ok, i removed it